src/sqlfluff/core/linter/linted_file.py

```diff
@@ -325,6 +333,19 @@ def fix_string(self) -> Tuple[Any, bool]:
 
             # Deal with the easy case of only literals
             if set(local_type_list) == {"literal"}:
+                # Is there leftover source following the patch?
```

Comment:
The changes in this section fixed the initial issue (code replaced by whitespace), but not the overall issue (e.g. duplicate code inside a loop). This change may still be useful in some circumstances, as it is not specific to templates or loops, so I recommend we keep it.